### PR TITLE
Add support for "include_segment_file_sizes" in indices stats REST handler

### PR DESCRIPTION
**Pull request by spiegela**
Fix for #21878

Adds `include_segment_file_sizes` to acceptable parameters in the `RestIndicesStatsAction` REST handler.

Thanks!

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestIndicesStatsAction.java [hunk 1]
```diff
@@ -157,7 +158,7 @@ public boolean canTripCircuitBreaker() {
         return false;
     }
 
-    private static final Set<String> RESPONSE_PARAMS = Collections.singleton("level");
+    private static final Set<String> RESPONSE_PARAMS = new HashSet<String>(Arrays.asList("level", "include_segment_file_sizes"));
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestIndicesStatsAction.java]: This isn't the right fix, it's not a response parameter (it controls the request that the node client sends). Rather, this parameter needs to be consumed (and parsed as a boolean).

**Review by jasontedor — CHANGES_REQUESTED** [review]

**Comment by jasontedor** [comment]
I left a comment requesting a different approach. The same fix needs to be applied to node stats too.

**Comment by jasontedor** [comment]
@spiegela Also, I think I'd prefer the tests to be REST tests as opposed to unit tests; please let me know if you can make these changes or if I can help guide you through this.

File: core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestIndicesStatsAction.java (hunk 1 above)
Comment (spiegela) [inline core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestIndicesStatsAction.java]: Gotcha. I see some examples of that now using the `request.paramAsBoolean`.  I'll correct it this way, and correct it in the node stats, too.

**Comment by spiegela** [comment]
Thanks, @jasontedor.  I can make the fix by consuming the parameter pretty quickly. Can you point me to a REST test example?

**Comment by jasontedor** [comment]
Look at `rest-api-spec/src/main/resources/rest-api-spec/test/indices.stats` and similarly for nodes stats. And you can poke around in the sibling folders for more examples. Feel free to ping upon any questions.

**Comment by spiegela** [comment]
@jasontedor I've fixed the issue by consuming the param as a bool, and switched the testing to the REST tests. I've also corrected the same issue in the node stats.

File: rest-api-spec/src/main/resources/rest-api-spec/api/indices.stats.json [hunk 2]
```diff
@@ -52,6 +52,11 @@
         "types" : {
           "type" : "list",
           "description" : "A comma-separated list of document types for the `indexing` index metric"
+        },
+        "include_segment_file_sizes": {
+          "type": "boolean",
+          "description": "Whether report the aggregated disk usage of each one of the Lucene index files.",
```
Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/api/indices.stats.json]: Can you fix the grammar here, remove the period, and add a remark that it only applies if segments stats are requested?

**Review by jasontedor — CHANGES_REQUESTED** [review]
Thanks @spiegela. Getting close.

**Comment by spiegela** [comment]
@jasontedor Should have waited till the morning to push and caught both of those. I also cleaned up some unused imports.

**Comment by jasontedor** [comment]
@spiegela I think you force pushed? The review process is easier if we go commit by commit and we can squash at the end on our side when we integrate. No worries this time, just a heads up for the future. :smile:

**Comment by jasontedor** [comment]
ok to test

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by jasontedor** [comment]
I will merge this soon.

**Comment by jasontedor** [comment]
Thanks @spiegela!

**Comment by spiegela** [comment]
Sure thing!

**Pull request merged**